tests/test_onion_requests.py

```diff
@@ -0,0 +1,289 @@
+from sogs.web import app
+from sogs import crypto
+from sogs.hashing import blake2b
+from sogs import utils
+from nacl.bindings import (
+    crypto_scalarmult,
+    crypto_aead_xchacha20poly1305_ietf_encrypt,
+    crypto_aead_xchacha20poly1305_ietf_decrypt,
+)
+from Cryptodome.Cipher import AES
+import nacl.utils
+import struct
+import json
+import auth
+
+from nacl.public import PrivateKey
+
+# ephemeral X25519 keypair for use in tests
+a = PrivateKey(bytes.fromhex('ec32fb5766cf52b1b5d7b0bff08e29f5c0c58ca19beaf6a5c7d3dd8ac7ced963'))
+A = a.public_key
+assert A.encode().hex() == 'd79a50b82ba8ca665f854382b42ba159efd16eef87409e97a8d07395b9492928'
+
+# For xchacha20 with use the libsodium recommended shared key of H(aB || A || B), where H(.) is
+# 32-byte Blake2B
+shared_xchacha20_key = blake2b(
+    (
+        crypto_scalarmult(a.encode(), crypto.server_pubkey_bytes),
```

Comment:
`crypto.server_pubkey_hash_bytes` ?

Reply:
Maybe I'm not understanding.  We're computing `H(aB || A || B)` here, and the server can compute the same value via `H(bA || A || B)`